infra/convert-demo.rkt

```diff
@@ -15,10 +15,10 @@
 (define (format-expr is-version-10 expr)
   (match expr
     [(? string?)
-     (let ([parsed (string->number expr)])
-       (if parsed
-           parsed
-           (raise (error "string that is not a num"))))]
+     (define parsed (string->number expr))
+     (unless parsed
+       (raise (error "string that is not a num")))
+     parsed]
```

Comment:
This kinda is subpar, `(or (string->number expr) (error ...))` would be ideal.

Reply:
Oh, interesting idea! I like that one. I'll give it a shot.